source/vir/src/ast.rs

```diff
@@ -106,6 +106,14 @@ pub enum TriggerAnnotation {
     Trigger(Option<u64>),
 }
 
+/// Operations on Ghost and Tracked
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ModeCoercion {
+    DerefMut,
+    BorrowMut,
+    Other,
```

Comment:
can you elaborate on what these coercions are (especially 'Other')?

Reply:
I added comments to this code.  It turned out `DerefMut` was dead code so I removed it.